### PR TITLE
Cleanup MachineDeployments when deleting a cluster

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: api/pkg/controller/cluster/deleting.go [hunk 1]
```diff
@@ -79,6 +79,16 @@ func (cc *Controller) deletingNodeCleanup(c *kubermaticv1.Cluster) (*kubermaticv
 		return nil, fmt.Errorf("failed to get cluster machine client: %v", err)
 	}
 
+	machineDeploymentList, err := machineClient.ClusterV1alpha1().MachineDeployments(metav1.NamespaceSystem).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list cluster MachineDeployments: %v", err)
+	}
+	if len(machineDeploymentList.Items) > 0 {
+		if err := machineClient.Cluster().MachineDeployments(metav1.NamespaceSystem).DeleteCollection(&metav1.DeleteOptions{}, metav1.ListOptions{}); err != nil {
+			return nil, fmt.Errorf("failed to delete cluster MachineDeployments: %v", err)
```
Comment (mrIncompetent) [inline api/pkg/controller/cluster/deleting.go]: Error message => Could not delete MachineDeployments in user cluster

File: api/pkg/controller/cluster/deleting.go [hunk 2]
```diff
@@ -79,6 +79,16 @@ func (cc *Controller) deletingNodeCleanup(c *kubermaticv1.Cluster) (*kubermaticv
 		return nil, fmt.Errorf("failed to get cluster machine client: %v", err)
 	}
 
+	machineDeploymentList, err := machineClient.ClusterV1alpha1().MachineDeployments(metav1.NamespaceSystem).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list cluster MachineDeployments: %v", err)
```
Comment (mrIncompetent) [inline api/pkg/controller/cluster/deleting.go]: Error message => Could not list MachineDeployments from user cluster

Comment (mrIncompetent) [reply api/pkg/controller/cluster/deleting.go]: Or: failed to....

To stick to the existing pattern

File: api/pkg/controller/cluster/deleting.go (hunk 1 above)
Comment (mrIncompetent) [reply api/pkg/controller/cluster/deleting.go]: Or: failed to....

To stick to the existing pattern

**Review by mrIncompetent — CHANGES_REQUESTED** [review]

File: api/pkg/controller/cluster/deleting.go (hunk 1 above)
Comment (alvaroaleman) [reply api/pkg/controller/cluster/deleting.go]: That is exactly what I did, if you look at the lines below, the error message there is `failed to list cluster machines: %v`

File: api/pkg/controller/cluster/deleting.go (hunk 2 above)
Comment (alvaroaleman) [reply api/pkg/controller/cluster/deleting.go]: That is exactly what I did, if you look at the lines below, the error message there is `failed to ldelete cluster machines: %v`

Comment (mrIncompetent) [reply api/pkg/controller/cluster/deleting.go]: I find the "cluster *" a bit confusing. We use "user cluser" everywhere else to express which cluster is affected

Comment (alvaroaleman) [reply api/pkg/controller/cluster/deleting.go]: Well I am fine with changing it but then we should change it anywhere. So I will just replace the `cluster` everywhere with `user cluster`?

**Comment by alvaroaleman** [comment]
@mrIncompetent PTAL, I have updated the error messages and changed the code, to wait for for MachineDeployments and MachineSets to be gone after deleting them

**Review by mrIncompetent — APPROVED** [review]

**Pull request merged**